### DISCUSSION TITLE
add environment variable to every container for a multi-container pod using rego

### DISCUSSION THREAD

**Discussion by zhoupoko2000** [open-policy-agent/community#80] [OPA and Rego] [answered]
Hello Team,
I am also working on it but I think since I am still learning, it is good to also ask the community so we learn from the best practice.

May I ask if any rego snippets that can be provided that to add environment variables to every container and init container in a multiple container pod?

thanks.

**Comment by srenatus** [comment]
Meta: I've converted this from an issue in OPA to a question here -- hope that's OK, I think _this_ is the better venue. I wasn't sure if the category was correct, though: Did you mean this as a general OPA/Rego question (this category) or a Gatekeeper question?

**Reply by zhoupoko2000** [reply]
@srenatus this is a opa core question. But I have solved it. no worries. But still thank you for the coordination here.

**Reply by srenatus** [reply]
Cool, can you post your solution here for everyone's benefit?

**Comment by zhoupoko2000** [comment, marked as answer]
absolutely. just give me some time. as this is from my work. so I need to extract the basic code snippet/gist and re-type everything here.

**Reply by zhoupoko2000** [reply]
for mutating webhook framework, @tsandall provided many examples, e.g. https://github.com/open-policy-agent/opa/issues/943
for the patches logic of this case, below is my solution.

`
patches[patch]{

    some i
    input.request.object.spec.containers[i]
    envPath := sprinf("/spec/containers/%s/env", [format_int(i, 10)])
    patch := [
    	{
        "op": "add",
        "patch": envPath,
        "value": [{
        	"name": "test_env",
                "value": "true"
        }]
        }
    ]
}
`
